flink-state-backends/flink-statebackend-heap-spillable/src/test/java/org/apache/flink/runtime/state/heap/SkipListKeyComparatorTest.java

```diff
@@ -0,0 +1,240 @@
+/*
+ *
+ *  * Licensed to the Apache Software Foundation (ASF) under one
+ *  * or more contributor license agreements.  See the NOTICE file
+ *  * distributed with this work for additional information
+ *  * regarding copyright ownership.  The ASF licenses this file
+ *  * to you under the Apache License, Version 2.0 (the
+ *  * "License"); you may not use this file except in compliance
+ *  * with the License.  You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package org.apache.flink.runtime.state.heap;
+
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+import org.apache.flink.api.common.typeutils.TypeSerializerSnapshot;
+import org.apache.flink.api.common.typeutils.base.IntSerializer;
+import org.apache.flink.api.common.typeutils.base.LongSerializer;
+import org.apache.flink.api.common.typeutils.base.TypeSerializerSingleton;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for {@link SkipListKeyComparator}.
+ */
+public class SkipListKeyComparatorTest extends TestLogger {
+
+	@Test
+	public void testCompareSkipListKeyForPrimitive() {
+		// The comparison of the serialized forms are identical to that of
+		// the values only when the numbers to compare are both not negative.
+		TypeSerializer<Long> keySerializer = LongSerializer.INSTANCE;
+		TypeSerializer<Integer> namespaceSerializer = IntSerializer.INSTANCE;
+
+		SkipListKeySerializer<Long, Integer> skipListKeySerializer =
+			new SkipListKeySerializer<>(keySerializer, namespaceSerializer);
+
+		// verify equal namespace and key
+		verifySkipListKey(skipListKeySerializer, 0L, 0, 0L, 0, 0);
```

Comment:
Nit: It is usually a good idea to not move the assertions too deep into helper methods. So instead of passing the expected result as the last argument, one could have a `compare` method which returns the comparison result and then one could check with an assertion on the top level whether it is correct. That way, the reader clearly sees what is asserted. Alternatively, one could give the verify method a bit more meaningful name saying wether it asserts that the first value is smaller or bigger than the second. Now one needs to know that the last value is the expected result.